Migrate THCState to ATen

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66391
* #66390
* #66389
* #66388
* __->__ #65948

This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

Differential Revision: [D31386275](https://our.internmc.facebook.com/intern/diff/D31386275)

peterbell10 committed:
Migrate THCState to ATen

This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCState to ATen"


This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCState to ATen"


This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCState to ATen"


This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31386275).

ngimel commented on aten/src/ATen/cuda/detail/CUDAHooks.cpp:
```diff
@@ -47,26 +48,28 @@ namespace detail {
 std::unique_ptr<THCState, void (*)(THCState*)> CUDAHooks::initCUDA() const {
   C10_LOG_API_USAGE_ONCE("aten.init.cuda");
-  THCState* thc_state = THCState_alloc();
+  THCState* thc_state = new THCState();
```
can you leave a comment here that thc_state is now an empty struct?

peterbell10 committed:
Update on "Migrate THCState to ATen"


This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

Differential Revision: [D31386275](https://our.internmc.facebook.com/intern/diff/D31386275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCState to ATen"


This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

Differential Revision: [D31386275](https://our.internmc.facebook.com/intern/diff/D31386275)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCState to ATen"


This guts `THCState` to simply be an empty struct, as well as:
- moving `THCState_getPeerToPeerAccess` and its cache into `ATen`.
- cleaning up dead code in `THCGeneral.cpp`
- moving `THCudaInit` and `THCMagma_init` into `CUDAHooks::initCUDA`

Differential Revision: [D31386275](https://our.internmc.facebook.com/intern/diff/D31386275)

[ghstack-poisoned]

suo commented:
This appears to have broken master, see: https://hud.pytorch.org/commit/a6774d6e1f3a88f931a3786b7c2617a74bdaa50e. I am reverting.

For some reason, the relevant windows + cuda tests did not run on your PR: https://github.com/pytorch/pytorch/actions/runs/1321834530. cc @zhouzhuojie @seemethere

malfet commented:
@peterbell10 when re-landing the PR, can you please add `ciflow/cuda` label to make sure that all CUDA tests are being run?

peterbell10 commented:
Yup, already seeing the test fail again in #66480.
